activesupport/lib/active_support/syntax_error_proxy.rb

```diff
@@ -32,6 +32,8 @@ def spot(_)
     end
 
     def backtrace_locations
+      return nil if super.nil?
```

Comment:
We should probably keep the return of `super` in a variable to avoid allocating a large array twice.

Reply:
I did check and the result seems to be memoised internally:

```
$ irb
irb(main):001> ex = begin raise "foo"; rescue => e; e; end
=> #<RuntimeError: foo>
irb(main):002> ex.backtrace_locations.__id__
=> 83100
irb(main):003> ex.backtrace_locations.__id__
=> 83100
```

We're also already calling it twice, once via `parse_message_for_trace` and once via `super`.

I'd be happy to fix that too, if I'm wrong and there actually is extra allocations happening.

Reply:
Here's where it's memoised:

https://github.com/ruby/ruby/blob/9f02680015d21013fa83587d8c912d2af621bb57/vm_backtrace.c#L776-L779

Reply:
👍  my bad. I'd have expected the array to be duped.